src/tree/updater_colmaker.cc

```diff
@@ -157,11 +157,11 @@ class ColMaker: public TreeUpdater {
             feat_index.push_back(i);
           }
         }
-        unsigned n = static_cast<unsigned>(param.colsample_bytree * feat_index.size());
+        unsigned n = std::max(static_cast<unsigned>(1),
+                              static_cast<unsigned>(param.colsample_bytree * feat_index.size()));
```

Comment:
To explain this bug, this will take colsample=0.4 with features=2 and return n=0 because of integer math.  It should round or at least (as here) be no smaller than 1.